starport/services/scaffolder/component.go

```diff
@@ -142,3 +150,83 @@ func isGoReservedWord(name string) bool {
 	}
 	return false
 }
+
+// ErrComponentAlreadyCreated is the error returned when a specific component is already created
+type ErrComponentAlreadyCreated struct {
+	name          string
+	checkedType   string
+	componentType string
+}
+
+// NewErrComponentAlreadyCreated returns a new ErrComponentAlreadyCreated error
+func NewErrComponentAlreadyCreated(name, checkedType, componentType string) *ErrComponentAlreadyCreated {
+	return &ErrComponentAlreadyCreated{
+		name,
+		checkedType,
+		componentType,
+	}
+}
+
+func (e *ErrComponentAlreadyCreated) Error() string {
+	return fmt.Sprintf("component %s with name %s is already created (type %s exists)",
+		e.componentType, e.name, e.checkedType)
+}
+
+// checkComponentCreated checks if the component has been already created with Starport in the project
+func checkComponentCreated(appPath, moduleName, compName string) (err error) {
+	compNameTitle := strings.Title(compName)
+
+	// associate the type to check with the component that scaffold this type
+	typesToCheck := map[string]string{
+		compNameTitle:                           componentType,
+		"MsgCreate" + compNameTitle:             componentType,
+		"MsgUpdate" + compNameTitle:             componentType,
+		"MsgDelete" + compNameTitle:             componentType,
+		"QueryAll" + compNameTitle + "Request":  componentType,
+		"QueryAll" + compNameTitle + "Response": componentType,
+		"QueryGet" + compNameTitle + "Request":  componentType,
+		"QueryGet" + compNameTitle + "Response": componentType,
+		"Msg" + compNameTitle:                   componentMessage,
+		"Query" + compNameTitle + "Request":     componentQuery,
+		"Query" + compNameTitle + "Response":    componentQuery,
+		"MsgSend" + compNameTitle:               componentPacket,
+		compNameTitle + "PacketData":            componentPacket,
+	}
+
+	absPath, err := filepath.Abs(filepath.Join(appPath, "x", moduleName, "types"))
+	if err != nil {
+		return err
+	}
+	fileSet := token.NewFileSet()
+	all, err := parser.ParseDir(fileSet, absPath, func(os.FileInfo) bool { return true }, parser.ParseComments)
+	if err != nil {
+		return err
+	}
+
+	for _, pkg := range all {
```

Comment:
https://github.com/tendermint/starport/blob/8863e892686bdff28ece8cf0c88db43a93764a4e/starport/services/scaffolder/component.go#L196-L232

Reply:
Why using a type `Type` not just using a string? `FindType(src string, names []string) string` or returning all the types, it will be just slightly less efficient

> Actually this API is finer because we don't duplicate names []string with returned []Type and eliminate the need of comparing both in case a given name not found and thus, not included to the returned slice:

I didn't get what you mean